### PR TITLE
Add RCTLinking to React.podspec

### PR DESCRIPTION
**Pull request by LeoLeBras**
Add RCTLinking to React.podspec.

### CODE REVIEW

File: React.podspec [hunk 1]
```diff
@@ -114,6 +114,12 @@ Pod::Spec.new do |s|
     ss.preserve_paths   = "Libraries/LinkingIOS/*.js"
   end
 
+  s.subspec 'RCTLinking' do |ss|
+    ss.dependency         'React/Core'
+    ss.source_files     = "Libraries/Linking/*.{h,m}"
```
Comment (grabbou) [inline React.podspec]: There is no native code living in Linking folder. Everything is in `LinkingIOS`.

**Comment by grabbou** [comment]
Shall we rename `RCTLinkingIOS` to `RCTLinking` ? If yes, I would either:
- define `RCTLinking` along and mark the `IOS` one as deprecated to be removed in next release so people can upgrade.
- mention in release notes

**Comment by nevir** [comment]
If there's no native code in `Libraries/Linking`, this doesn't seem to make much sense as a cocoapods subspec, I think?

**Comment by facebook-github-bot-4** [comment]
@LeoLeBras do you have any updates for this pull request? It's been a while so we wanted to check in and see if you've looked at the requested changes.

**Comment by javache** [comment]
Hasn't seen activity for a while. Feel free to send a new PR if this is still required.

**Pull request closed without merging**